docs/plugins/npe2_manifest_specification.md

```diff
@@ -0,0 +1,606 @@
+(npe2-manifest-spec)=
+
+# npe2 manifest specification
+
+```{admonition} Backward compatibility
+:class: tip
+We introduced a new plugin engine in December 2021. The new library [`npe2`][npe2]
+is a re-imagining of how napari interacts with plugins. Plugins targeting
+`napari-plugin-engine` will continue to work, but we recommend migrating to
+`npe2` as soon as possible.  See the [](npe2-migration-guide).
+```
+
+The **plugin manifest** is a specially formatted text file declaring the
+functionality of a [npe2][] plugin. A **plugin** is a python package that
+contains the manifest together with a suitable _[entry point group][epg]_ in
+the package metadata.
+
+Manifest files may be [json][], [yaml][], or [toml][] files conforming to the
+manifest schema. The **schema** defines what to expect in a manifest by
+defining the fields and their data types. These fields and their meanings are
+described below.
+
+A **plugin engine** is used to discover plugins, provide utilities for
```

Comment:
they don't need to know this when writing the manifest

Reply:
I do refer to the "plugin engine" later on. But I don't refer to discovery.  Should I remove the paragraph? It looks like a pretty reasonable idea. I could move the definition of plugin engine down to where it's used.